### PR TITLE
Fire discordsrv events when processing venture chat messages from VentureChatEvent

### PR DESCRIPTION
**Pull request by LOOHP**
This PR adds `VentureChatMessagePreProcessEvent` and `VentureChatMessagePostProcessEvent` to the `VentureChatHook` to allow other plugins to modify how venture chat messages look like before setting to discord. (Like how `GameChatMessagePreProcessEvent` and `GameChatMessagePostProcessEvent` works)

### CODE REVIEW

File: src/main/java/github/scarsz/discordsrv/api/events/VentureChatMessagePostProcessEvent.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.bukkit.event.Cancellable;
+
+import lombok.Getter;
+import lombok.Setter;
+import mineverse.Aust1n46.chat.api.events.VentureChatEvent;
+
+/**
+ * <p>Called after DiscordSRV has processed a VentureChat message but before being sent to Discord.
+ * Modification is allow and will effect the message sent to Discord.</p>
+ */
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/api/events/VentureChatMessagePostProcessEvent.java:34]: This should specify that it's a VentureChat Bungee message when the VentureChatBungee config option and not just any VentureChat message (something like "... VentureChat bungee message (when the VentureChatBungee config option is enabled) ...") is enabled. 

I also think the regular -PreProcess and -PostProcess events should have a note about these new events (something like "VentureChat bungee messages (when the VentureChatBungee config option is enabled) will use the VentureChatMessage...ProcessEvent instead due to the lack of the Player object")

File: src/main/java/github/scarsz/discordsrv/hooks/chat/VentureChatHook.java [hunk 2]
```diff
@@ -153,6 +163,14 @@ public void onVentureChat(VentureChatEvent event) {
             discordMessage = discordMessage.replace("@", "@\u200B"); // zero-width space
             message = message.replace("@", "@\u200B"); // zero-width space
         }
+        
+        VentureChatMessagePostProcessEvent postEvent = DiscordSRV.api.callEvent(new VentureChatMessagePostProcessEvent(channel, discordMessage, event, preEvent.isCancelled()));
+        if (postEvent.isCancelled()) {
+        	DiscordSRV.debug(Debug.MINECRAFT_TO_DISCORD, "VentureChatMessagePostProcessEvent was cancelled, message send aborted");
+            return;
+        }
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/hooks/chat/VentureChatHook.java:171]: Bad indentation here
```suggestion
        if (postEvent.isCancelled()) {
            DiscordSRV.debug(Debug.MINECRAFT_TO_DISCORD, "VentureChatMessagePostProcessEvent was cancelled, message send aborted");
            return;
        }
```

**Comment by LOOHP** [comment]
Resolved the indentation and updated documentation

**Review by Vankka — APPROVED** [review]

**Comment by LOOHP** [comment]
Any TBA on when will PRs make it into the snapshots?

**Comment by Vankka** [comment]
Hi, sorry about this PR taking so long and the lack of responses (I saw your ticket, too just didn't get a chance to respond). 

We don't really have a timeline or even a standardized way to review pull requests, so how they're reviewed and merged may vary person to person. I apologize that this one has taken such a long time to get merged, I haven't been as active on DiscordSRV (v1 specifically) recently so that has only made it worse.

I'll merge this now, seeing as nobody else has had anything to comment about since I approved this. Hope you have a nice day!

**Comment by LOOHP** [comment]
Thank you for merging and the clarification :)

**Pull request merged**